### ISSUE TITLE
Switch to Puppetlabs Docker Module

### ISSUE THREAD

**Issue by martialblog** [vision-it/vision-docker#3]
Puppetlabs has released a new Docker Module. The development on the garethr-docker Module seems to be stalled.

https://github.com/puppetlabs/puppetlabs-docker

**Comment by martialblog** [comment]
Started working on this: https://github.com/vision-it/vision-docker/tree/puppetmodule

Needs futher testing

**Comment by martialblog** [comment]
- [x] Extend spec/beaker tests
- [x] Test in Vagrant
- [x] Test change from old module to new in Vagrant

**Comment by martialblog** [comment]
@jacksgt The Debian 9 Beaker test fails... gotta fix that

**Comment by jacksgt** [comment]
Contrary to the `debian:jessie` base image, `debian:stretch` no longer has `systemd` preinstalled, to save space, because usually one doesn't need another process supervisor inside a Docker container.

Nevertheless, Beaker tries to verify that the service is running (inside the container):

https://github.com/vision-it/vision-docker/blob/0930a8e1b25f6e03def1c0b3841308a60b2c780c/spec/classes/compile_spec.rb#L24
```
Error: /Stage[main]/Docker::Service/Exec[docker-systemd-reload-before-service]: Failed to call refresh: 'systemctl daemon-reload > /dev/null' returned 1 instead of one of [0]
...
Error: /Stage[main]/Docker::Service/Service[docker]: Provider systemd is not functional on this host
```

**Comment by jacksgt** [comment]
Fixed in https://github.com/vision-it/vision-docker/pull/5

**Comment by martialblog** [comment]
@jacksgt The change in debian9.yaml should be included into the modulesync... otherwise it'll be overwritten

**Comment by martialblog** [comment]
Done https://github.com/vision-it/modulesync_configs/commit/bd49366bee572286719ee9c60e6a0b77acbad4fc

**Comment by martialblog** [comment]
Fixed in #7
